sklearn/naive_bayes.py

```diff
@@ -117,6 +118,320 @@ def predict_proba(self, X):
+        else:
+            raise ValueError("Class priors for every estimator "
+                             "must be the same.")
+
+        # Obtain the jll of each fitted estimator
+        jlls = [nb_model._joint_log_likelihood(
+                    np.array(_safe_indexing(X, cols, axis=1)))
+                for (_, nb_model, cols) in self.models_]
+
+        # Stack these jlls to give us
+        # the shape (estimator, sample, class)
+        jlls = np.hstack([jlls])
+
+        # Subtract the class log prior from all the jlls
+        # but add it back after the summation
+        jlls = jlls - log_prior
+        jll = jlls.sum(axis=0) + log_prior
+
+        return jll
+
+    @property
+    def _models(self):
+        return [(name, model) for name, model, _ in self.models]
+
+    @_models.setter
+    def _models(self, value):
+        self.models = [
+            (name, nb_model, cols) for ((name, nb_model), (_, _, cols))
+            in zip(value, self.models)]
+
+    def get_params(self, deep=True):
+        return self._get_params('_models', deep=deep)
+
+    def set_params(self, **kwargs):
+        self._set_params('_models', **kwargs)
+        return self
+
+    def _validate_models(self, X):
+
+        valid_modules = copy.copy(__all__)
+        valid_modules.remove("GeneralNB")
+
+        self._cols = []
+        dict_col2model = {}
+        all_fit_priors = []
+        all_class_priors = []
+
+        # Check type of `models` parameter
+        if not isinstance(self.models, list):
+            raise TypeError(
+                "Expected list but got {}".format(type(self.models)))
+
+        names, _, _ = zip(*self.models)
+        self._validate_names(names)
+
+        for model in self.models:
+
+            # Check type of each entry in list
+            if not isinstance(model, tuple):
+                raise TypeError(
+                    "Expected list of tuples "
+                    "but got list of {}s".format(type(model)))
+            if len(model) != 3:
+                raise ValueError("Expected tuple to have length of 3 "
+                                 "but got {}".format(len(model)))
+
+            _, estimator, cols = model
+
+            # Check naive bayes estimator for format
+            # `fit` and `_joint_log_likelihood` attributes
+            if callable(estimator):
+                raise ValueError("Wrong format specified.")
+            if not (hasattr(estimator, "fit")
+                    or hasattr(estimator, "_joint_log_likelihood")):
+                raise TypeError("Naive bayes estimator should implement "
+                                "the fit and _joint_log_likelihood methods. "
+                                "{} doesn't.".format(type(estimator)))
+            if estimator.__class__.__name__ not in valid_modules:
+                raise ValueError(
+                    "Distributions should be one of {}".format(valid_modules))
+
+            # Check naive bayes estimator for attributes
```

Comment:
Sorry, which parameters are you referring to?

Reply:
Sorry, how do you mean? Could you please explain a little bit on this?

Reply:
I mean GeneralNB could have a `class_prior` and a `fit_prior` param

Reply:
Oh I think that's a good idea because we can use these 2 parameters to override all `class_prior`s and `fit_prior`s that the user might have defined in the other naive Bayes models. In this way, we could remove several checks that tries to make sure all of these priors must be the same. I'll proceed with this.